kfs-core/src/main/java/edu/arizona/kfs/pdp/service/impl/PdpEmailServiceImpl.java

```diff
@@ -0,0 +1,201 @@
+        try {
+            mailService.sendMessage(message);
+        }
+        catch (Exception e) {
+        	 LOG.error("sendPrepaidChecksLoadEmail() Invalid email address. Message not sent", e);
+             throw new RuntimeException("sendPrepaidChecksLoadEmail() Invalid email address. Message not sent " + e.getMessage(), e);             
+        }        
+
+    }
+
+    public void sendPrepaidChecksErrorEmail(PaymentFileLoad prepaidChecksFile, MessageMap errors, String fileName) {    
+        LOG.debug("sendPrepaidChecksErrorEmail() starting");
+
+        // check email configuration
+        if (!super.isPaymentEmailEnabled()) {
+            return;
+        }
+
+        MailMessage message = new MailMessage();
+
+        String returnAddress = parameterService.getParameterValueAsString(KfsParameterConstants.PRE_DISBURSEMENT_BATCH.class, KFSConstants.FROM_EMAIL_ADDRESS_PARM_NM);
+        if(StringUtils.isEmpty(returnAddress)) {
+            returnAddress = mailService.getBatchMailingList();
+        }
+        message.setFromAddress(returnAddress);
+        message.setSubject("Prepaid Checks File Load ERROR Notification - " + fileName);
+
+        StringBuilder body = new StringBuilder();
+        List<String> ccAddresses = new ArrayList<String>( parameterService.getParameterValuesAsString(LoadPaymentsStep.class, PdpParameterConstants.HARD_EDIT_CC) );
+
+        if (prepaidChecksFile == null) {
+            if (ccAddresses.isEmpty()) {
+                LOG.warn("sendPrepaidChecksErrorEmail() No HARD_EDIT_CC addresses.  No email sent");
+                return;
+            }
+
+            message.getToAddresses().addAll(ccAddresses);
+
+            body.append(getMessage(PdpKeyConstants.MESSAGE_PAYMENT_EMAIL_BAD_FILE_PARSE) + "\n\n");
+        }
+        else {       	
+            CustomerProfile customer = customerProfileService.get(prepaidChecksFile.getChart(), prepaidChecksFile.getUnit(), prepaidChecksFile.getSubUnit());
+            if (customer == null) {
+                LOG.error("sendPrepaidChecksErrorEmail() Invalid Customer.  Sending email to CC addresses");
+
+                if (ccAddresses.isEmpty()) {
+                    LOG.error("sendPrepaidChecksErrorEmail() No HARD_EDIT_CC addresses.  No email sent");
+                    return;
+                }
+
+                message.getToAddresses().addAll(ccAddresses);
+
+                body.append(getMessage(PdpKeyConstants.MESSAGE_PAYMENT_EMAIL_INVALID_CUSTOMER) + "\n\n");
+            }
+            else {           	
+                String toAddresses = StringUtils.deleteWhitespace(customer.getProcessingEmailAddr());
+                List<String> toAddressList = Arrays.asList(toAddresses.split(","));
+
+                message.getToAddresses().addAll(toAddressList);
+                message.getCcAddresses().addAll(ccAddresses);
+                //TODO: for some reason the mail service does not work unless the bcc list has addresss. This is a temporary workaround

```

Comment:
This TODO was in both the KFS 3.0 and KFS 6.0 code. Pulled from parent class in KFS 6.0, so.... I didn't really want to remove the code and take a chance on this.